### ISSUE TITLE
Different data source for zoom modal.

### ISSUE THREAD

**Issue by JCMais** [kartik-v/bootstrap-fileinput#739]
Is there an option to display a different source on the modal that appears when you click on the zoom button?

I'm asking that because I wanted to show just a thumbnail in the initial preview element, since the images are pretty big, showing all of them at full size would be really bad for performance.

**Comment by kartik-v** [comment]
Will need to be an enhancement... you can submit a PR to accelerate this.

**Comment by JCMais** [comment]
I think this can be closed, unless you want to add that feature.

I was able to do what I wanted by just using the ``filezoomshow`` event (thank you for having some many events available):

```
//...
.on( 'filezoomshow', function( event, params ) {

            var $filePreviewImage = params.modal.find( '.file-preview-image' ),
                  src = $filePreviewImage.prop( 'src' );

            // not an image
            if ( !$filePreviewImage.length )
                return;

            //In my case, thumbnail src looks like: picture-thumb-{reallylongid}.png
            src = src.replace( 'picture-thumb', 'picture' );

            $filePreviewImage.prop( 'src', src );
        });
```

**Comment by mrpotts** [comment]
I need this feature as well, but the 'filezoomshow' event doesn't seem to be triggered. I copy/pasted the example in the documentation: 
```
$('#input-id').on('filezoomshow', function(event, params) {
    console.log('File zoom show ', params.sourceEvent, params.previewId, params.modal);
});
```
...but it's not doing anything. None of the _filezoom_ events works, except filezoomnext and filezoomprev. I'm using v4.3.9.

**Comment by mrpotts** [comment]
I think I've found the problem. The _initZoom function is:
```
_initZoom: function () {
  ...
  $.each($h.MODAL_EVENTS, function (event) {
    self._listenModalEvent(event);
  });
```
But it should be:
```
  self._listenModalEvent( $h.MODAL_EVENTS[event] );
```
because __listenModalEvent_ espects a string, not an integer.

**Comment by kartik-v** [comment]
> because _listenModalEvent espects a string, not an integer

Fixed via d185163.

**Comment by JCMais** [comment]
this has been fixed a long time ago, closing.
